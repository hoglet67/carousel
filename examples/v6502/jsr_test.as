        .ORG $8000

        LDA $00
        TAX

TOP:    
        NOP
        JSR SUB1
        JSR SUB2
        JSR SUB3
        NOP
        JMP TOP
        
        .WORD 00,00,00,00
        .WORD 00,00,00,00
        
SUB1:
        LDA #'6'
        STA $0000,X
        RTS

SUB2:
        LDA #'5'
        INX
        STA $0000,X
        RTS

SUB3:
        JSR SUB4
        LDA #'2'
        INX
        STA $0000,X
        INX
        RTS

SUB4:
        LDA #'0'
        INX
        STA $0000,X
        RTS




