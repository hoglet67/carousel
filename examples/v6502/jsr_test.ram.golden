  
0000: 36 35 30 32 36 35 30 32 36 35 30 ff ff ff ff ff   65026502650.....
0010: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff   ................
0020: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff   ................
0030: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff   ................
0040: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff   ................
0050: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff   ................
0060: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff   ................
0070: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff   ................
